wqflask/wqflask/correlation/show_corr_results.py

```diff
@@ -484,30 +484,31 @@ def generate_corr_json(corr_results, this_trait, dataset, target_dataset, for_ap
         if trait.view == False:
             continue
         results_dict = {}
-        if not for_api:
-            results_dict['checkbox'] = "<INPUT TYPE='checkbox' NAME='searchResult' class='checkbox trait_checkbox' style='padding-right: 0px;' VALUE='" + hmac.hmac_creation('{}:{}'.format(trait.name, trait.dataset.name)) + "'>"
-            results_dict['index'] = i + 1
-            results_dict['trait_id'] = "<a href='/show_trait?trait_id="+str(trait.name)+"&dataset="+str(dataset.name)+"'>"+str(trait.name)+"</a>"
-        else:
-            results_dict['trait_id'] = trait.name
+        results_dict['index'] = i + 1
+        results_dict['trait_id'] = trait.name
+        results_dict['dataset'] = trait.dataset.name
         if target_dataset.type == "ProbeSet":
             results_dict['symbol'] = trait.symbol
+            if len(trait.description_display) > 40:
+                results_dict['description'] = trait.description_display[:40] + "..."
+            else:
+                results_dict['description'] = trait.description_display
             results_dict['description'] = trait.description_display
             results_dict['location'] = trait.location_repr
-            results_dict['mean'] = float(trait.mean)
+            if trait.mean and trait.mean != "":
+                results_dict['mean'] = float(trait.mean)
+            else:
+                results_dict['mean'] = "N/A"
             if trait.LRS_score_repr != "N/A":
                 results_dict['lrs_score'] = "%.1f" % float(trait.LRS_score_repr)
             else:
                 results_dict['lrs_score'] = "N/A"
             results_dict['lrs_location'] = trait.LRS_location_repr
-            if trait.additive != "":
+            if trait.additive and trait.additive != "":
                 results_dict['additive'] = "%0.3f" % float(trait.additive)
             else:
                 results_dict['additive'] = "N/A"
-            if for_api:
-                results_dict['sample_r'] = "%0.3f" % float(trait.sample_r)
-            else:
-                results_dict['sample_r'] = "<a target='_blank' href='corr_scatter_plot?dataset_1=" + str(dataset.name) + "&dataset_2=" + str(trait.dataset.name) + "&trait_1=" + str(this_trait.name) + "&trait_2=" + str(trait.name) + "'>" + "%0.3f" % float(trait.sample_r) + "</a>"
+            results_dict['sample_r'] = "%0.3f" % float(trait.sample_r)
```

Comment:
**suggestion:nitpick:** Since we now use Python3, it's a nice idea to switch to [f-strings](http://zetcode.com/python/fstring):

```python
#+begin_src python :results output
float_string = "0.35535115"
print("%0.3f" % float(float_string))

print(f"{float(float_string):.3f}")
#+end_src

#+RESULTS:
: 0.355
: 0.355
```

So this would be:
```suggestion
            results_dict['sample_r'] = f"{float(trait.sample_r):.3f}"
```
Idea is to not do this *at* once. Instead, when we encounter a %-string(or when you need formatted strings), we simply default to f-strings.
cc @Alexanderlacuna